src/core/qgsogcutils.cpp

```diff
@@ -312,10 +321,29 @@ QgsGeometry QgsOgcUtils::geometryFromGMLPolygon( const QDomElement &geometryElem
     {
       return QgsGeometry();
     }
-    if ( readGMLPositions( exteriorPointList, posElement ) != 0 )
+    if ( readGMLPositions( exteriorPointList, posElement ) != false )
```

Comment:
```suggestion
    if ( readGMLPositions( exteriorPointList, posElement ) )
```

Reply:
I asked myself the same question: why the original author used ` != 0` ?
I think that he wanted to capture reader's attention to the counter-intuitive fact that `readGMLPositions` result `true` is the error condition and `false` means that all was ok.

I chose to leave that (and just use `false` instead of `0` for readability), I think that the compiler would be smart enough to generate the same code if omitting the comparison.